pandas/core/arrays/interval.py

```diff
@@ -0,0 +1,1024 @@
+
+        formatter = str
+
+        if n == 0:
+            summary = '[]'
+        elif n == 1:
+            first = formatter(self[0])
+            summary = '[{first}]'.format(first=first)
+        elif n == 2:
+            first = formatter(self[0])
+            last = formatter(self[-1])
+            summary = '[{first}, {last}]'.format(first=first, last=last)
+        else:
+
+            if n > max_seq_items:
+                n = min(max_seq_items // 2, 10)
+                head = [formatter(x) for x in self[:n]]
+                tail = [formatter(x) for x in self[-n:]]
+                summary = '[{head} ... {tail}]'.format(
+                    head=', '.join(head), tail=', '.join(tail))
+            else:
+                head = []
+                tail = [formatter(x) for x in self]
+                summary = '[{tail}]'.format(tail=', '.join(tail))
+
+        return summary
+
+    def __repr__(self):
+        tpl = textwrap.dedent("""\
+        {cls}({data},
+        {lead}closed='{closed}',
+        {lead}dtype='{dtype}')""")
+        return tpl.format(cls=self.__class__.__name__,
+                          data=self._format_data(),
+                          lead=' ' * len(self.__class__.__name__) + ' ',
+                          closed=self.closed, dtype=self.dtype)
+
+    def _format_space(self):
+        space = ' ' * (len(self.__class__.__name__) + 1)
+        return "\n{space}".format(space=space)
+
+    @property
+    def left(self):
+        """
+        Return the left endpoints of each Interval in the IntervalArray as
+        an Index
+        """
+        return self._left
+
+    @property
+    def right(self):
+        """
+        Return the right endpoints of each Interval in the IntervalArray as
+        an Index
+        """
+        return self._right
+
+    @property
+    def closed(self):
+        """
+        Whether the intervals are closed on the left-side, right-side, both or
+        neither
+        """
+        return self._closed
+
+    _interval_shared_docs['set_closed'] = """
+        Return an %(klass)s identical to the current one, but closed on the
+        specified side
+
+        .. versionadded:: 0.24.0
+
+        Parameters
+        ----------
+        closed : {'left', 'right', 'both', 'neither'}
+            Whether the intervals are closed on the left-side, right-side, both
+            or neither.
+
+        Returns
+        -------
+        new_index : %(klass)s
+
+        Examples
+        --------
+        >>>  index = pd.interval_range(0, 3)
+        >>>  index
+        %(klass)s([(0, 1], (1, 2], (2, 3]]
+              closed='right',
+              dtype='interval[int64]')
+        >>>  index.set_closed('both')
+        %(klass)s([[0, 1], [1, 2], [2, 3]]
+              closed='both',
+              dtype='interval[int64]')
+        """
+
+    @Appender(_interval_shared_docs['set_closed'] % _shared_docs_kwargs)
+    def set_closed(self, closed):
```

Comment:
Should we make `IntervalArray.closed` a setter as well?

Reply:
I don't think this would be a problem with the current implementation, but #19371 makes me a bit hesitant since it'd make a `closed` setter a (limited) `dtype` setter.